chia/full_node/mempool_check_conditions.py

```diff
@@ -329,13 +330,18 @@ def get_name_puzzle_conditions(
                 if max_cost < 0:
                     return NPCResult(uint16(Err.INVALID_BLOCK_COST.value), [], uint64(0))
                 if cvl is not None:
-                    conditions_list.append(cvl)
+                    if cvl.opcode not in conditions_dict:
+                        conditions_dict[cvl.opcode] = set()
+                    elif cvl.opcode == ConditionOpcode.CREATE_COIN and cvl in conditions_dict[cvl.opcode]:
+                        # CREATE_COIN conditions are special. We don't allow
+                        # duplicate coins. Attempting to create two identical
+                        # coins is a consensus failure.
+                        return NPCResult(uint16(Err.DUPLICATE_OUTPUT.value), [], uint64(0))
+
+                    conditions_dict[cvl.opcode].add(cvl)
 
-            conditions_dict = conditions_by_opcode(conditions_list)
-            if conditions_dict is None:
-                conditions_dict = {}
             npc_list.append(
-                NPC(spent_coin.name(), spent_coin.puzzle_hash, [(a, b) for a, b in conditions_dict.items()])
+                NPC(spent_coin.name(), spent_coin.puzzle_hash, [(a, list(b)) for a, b in conditions_dict.items()])
```

Comment:
As the `.items()` value is already an iterators of pairs, the old version 
`[(a, b) for a, b in conditions_dict.items()]`
 was was actually identical to just 
`list(conditions_dict.items())`.
(Now you're left with the items `{a : list(b) for a, b in conditions_dict.items()}.items()`, which might lose its appeal if it's cast to list at the end. That said, the casting things to lists before consuming them might be unnecessary in many places, particularly if the resources is used only for another loop.